runtime/lua/vim/lsp/log.lua

```diff
@@ -32,6 +32,17 @@ do
 
   vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
   local logfile = assert(io.open(logfilename, "a+"))
+
+	local log_info = vim.loop.fs_stat(logfilename)
+	if log_info and log_info.size > 1e9 then
+		local warn_msg = string.format(
+			"Language server client log at %s has reached %d KB in size",
+			logfilename,
+			log_info.size / 1000
+		)
+		vim.notify(warn_msg)
+	end
```

Comment:
indentation looks weird.

> I went ahead and added a runtime check, but we can remove it (I personally would)

why remove it? it is likely unknown to many users, and unwanted to most of them.

Reply:
The runtime check? I just don't like adding things that are done automatically during the startup process. It's just an fstat, so probably not a big deal.

Reply:
definitely a good instinct, but this seems important enough (potentially catastrophic) to warrant some sort of active check. Doing it on (lazy) startup is a one-time cost so not too bad.